lib/transformations/utils.js

```diff
@@ -0,0 +1,169 @@
+function safeTraverse(obj, paths) {
+	let val = obj;
+	let idx = 0;
+
+	while (idx < paths.length) {
+		if (!val) {
+			return null;
+		}
+		val = val[paths[idx]];
+		idx++;
+	}
+	return val;
+}
+
+// Convert nested MemberExpressions to strings like webpack.optimize.DedupePlugin
+function memberExpressionToPathString(path) {
+	if (path && path.object) {
+		return [memberExpressionToPathString(path.object), path.property.name].join('.');
+	}
+	return path.name;
+}
+
+// Convert Array<string> like ['webpack', 'optimize', 'DedupePlugin'] to nested MemberExpressions
+function pathsToMemberExpression(j, paths) {
+	if (!paths.length) {
+		return null;
+	} else if (paths.length === 1) {
+		return j.identifier(paths[0]);
+	} else {
+		const first = paths.slice(0, 1);
+		const rest = paths.slice(1);
+		return j.memberExpression(
+			pathsToMemberExpression(j, rest),
+			pathsToMemberExpression(j, first)
+		);
+	}
+}
+
+/*
+* @function findPluginsByName
+*
+* Find paths that match `new name.space.PluginName()` for a given array of plugin names
+*
+* @param j — jscodeshift API
+* @param { Node } node - Node to start search from
+* @param { Array<String> } pluginNamesArray - Array of plugin names like `webpack.optimize.LoaderOptionsPlugin`
+* @returns Path
+ * */
+function findPluginsByName(j, node, pluginNamesArray) {
+	return node
+		.find(j.NewExpression)
+		.filter(path => {
+			return pluginNamesArray.some(
+				plugin => memberExpressionToPathString(path.get('callee').value) === plugin
+			);
+		});
+}
+
+/*
+ * @function findPluginsRootNodes
+ *
+ * Finds the path to the `plugins: []` node
+ *
+ * @param j — jscodeshift API
+ * @param { Node } node - Node to start search from
+ * @returns Path
+ * */
+function findPluginsRootNodes(j, node) {
+	return node.find(j.Property, { key: { name: 'plugins' } });
+}
+
+/*
+ * @function createProperty
+ *
+ * Creates an Object's property with a given key and value
+ * Note: For now it always create a literal (String) for keys.
+ *
+ * @param j — jscodeshift API
+ * @param { string | number } key - Property key
+ * @param { string | number | boolean } value - Property value
+ * @returns Node
+ * */
+function createProperty(j, key, value) {
+	return j.property(
+		'init',
+		j.literal(key),
```

Comment:
This has to be `j.identifier(key)`. Will change this and also related snapshots.